eos/utils/expression.cc

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2021 Méril Reboud
+ *
+ * This file is part of the EOS project. EOS is free software;
+ * you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#include <eos/utils/expression.hh>
+
+//#include <iostream>
+
+namespace eos::exp
+{
+    double BinaryExpression::Add(double a, double b)      { return a+b; }
+    double BinaryExpression::Subtract(double a, double b) { return a-b; }
+    double BinaryExpression::Multiply(double a, double b) { return a*b; }
+    double BinaryExpression::Divide(double a, double b)   { return a/b; }

```

Comment:
Come to think of it: ``add``, ``subtract``, ``multiply`` and ``divide`` are not expressions, they are actions. The expressions are ``sum``, ``difference``, ``product`` and ``ratio``. Care to change that?